### PR TITLE
[stable10] select2 mail input

### PR DESCRIPTION
**Pull request by lordelix**
## Description

- input field is adaptive (auto-scale)
- mail addresses become "fixed" by hitting space as in the screenshot
- mail addresses can be "unfixed"/deleted letter-wise with backspace and have a "x" to delete them at once

## Related Issue

https://github.com/owncloud/core/issues/29695
Fixes: https://github.com/owncloud/core/pull/30645

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

**Review by PVince81 — APPROVED** [review]
:+1:

**Comment by felixheidecke** [comment]
```
npm ERR! code ENOPACKAGEJSON
npm ERR! package.json Non-registry package missing package.json: @bower_components/Snap.js@github:jakiestfu/Snap.js#1.9.3.
npm ERR! package.json npm can't find a package.json file in your current directory.
```

Either abandon 5 year old Snap.js or downgrade npm.  :collision:

**Comment by DeepDiver1975** [comment]
> Either abandon 5 year old Snap.js or downgrade npm. boom

or adopt your PR to use bower and not yarn/npm

stable10 is still using bower ....

**Comment by felixheidecke** [comment]
Hmmm… looks like my cherry-pick wen't wrong

File: build/package.json [hunk 1]
```diff
@@ -9,21 +9,52 @@
   "private": true,
   "homepage": "https://github.com/owncloud/",
   "contributors": [],
-  "dependencies": {},
+  "dependencies": {
+    "@bower_components/Jcrop": "tapmodo/Jcrop#2.0.4",
```
Comment (PVince81) [inline build/package.json]: please adjust to use the non-yarn way on this branch and also retest

**Comment by PVince81** [comment]
basically just take the package.json that is already on stable10 and increase the select2 version, no need to change the whole file like on master

File: build/package-lock.json [hunk 2]
```diff
@@ -0,0 +1,6160 @@
+{
+  "name": "owncloud-js-dev",
```
Comment (PVince81) [inline build/package-lock.json]: not sure if it is wise to add this file right now while we didn't have this before ? please confirm

Comment (felixheidecke) [reply build/package-lock.json]: This seems to come from a commit 2 years ago: https://github.com/owncloud/core/blame/ec523e4266d74727d8e40182daf15861b9366f9c/build/package.json#L2

Comment (felixheidecke) [reply build/package-lock.json]: Lockfiles are generated automaticly in node 8 and they recommend to add it to the repository.
It prevents builds from pull newer / other libs than those we worked and tested with. So I see this as a contribution to solidity.

**Pull request merged**